File: server/router/control/file_manage.go
package control

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type FileManageRouter struct {
}

// InitFileManageRouter 初始化 FileManage表 路由信息
func (s *FileManageRouter) InitFileManageRouter(Router *gin.RouterGroup) {
	fileManageRouter := Router.Group("fileManage").Use(middleware.OperationRecord())
	fileManageRouterWithoutRecord := Router.Group("fileManage")
	var fileManageApi = v1.ApiGroupApp.ControlApiGroup.FileManageApi
	{
		fileManageRouter.POST("uploadFileChunk", fileManageApi.UploadFileChunk) //切片上传文件
		fileManageRouter.GET("abortFileUpload", fileManageApi.AbortFileUpload)  //取消切片上传
		fileManageRouter.PUT("option", fileManageApi.Option)                    // 配置管理
		fileManageRouter.DELETE("deleteItem", fileManageApi.DeleteItem)         // 删除文件
		fileManageRouter.PUT("renameItem", fileManageApi.RenameItem)            // 重命名文件
		fileManageRouter.PUT("copyItem", fileManageApi.CopyItem)                // 复制文件
		fileManageRouter.POST("createDirectory", fileManageApi.CreateDirectory) // 新建文件夹

	}
	{
		fileManageRouterWithoutRecord.POST("getFileItems", fileManageApi.GetFileItems) // 获取目录下文件
		fileManageRouterWithoutRecord.POST("downloadItem", fileManageApi.DownloadItem) // 新建文件夹
	}
}

File: web/src/api/fileManage.js
import service from '@/utils/request'

// 获取文件列表
export const getFileItems = (data) => {
  return service({
    url: '/fileManage/getFileItems',
    method: 'post',
    data
  })
}

// 配置修改
export const upOption = (data) => {
  return service({
    url: '/fileManage/option',
    method: 'put',
    data
  })
}

// 上传文件
export const uploadFileChunk = (data) => {
  return service({
    url: '/fileManage/uploadFileChunk',
    method: 'post',
    donNotShowLoading: true,
    headers: { 'Content-Type': 'multipart/form-data' },
    data
  })
}

// 取消上传
export const abortFileUpload = (params) => {
  return service({
    url: '/fileManage/abortFileUpload',
    method: 'get',
    params
  })
}

// 删除文件
export const deleteItem = (params) => {
  return service({
    url: '/fileManage/deleteItem',
    method: 'delete',
    params
  })
}

// 重命名
export const renameItem = (data) => {
  return service({
    url: '/fileManage/renameItem',
    method: 'put',
    data
  })
}
// 创建文件夹
export const createDirectory = (data) => {
  return service({
    url: '/fileManage/createDirectory',
    method: 'post',
    data
  })
}
// 复制文件
export const copyItem = (data) => {
  return service({
    url: '/fileManage/copyItem',
    method: 'put',
    data
  })
}

// 复制文件
export const downloadItem = (data) => {
  return service({
    url: '/fileManage/downloadItem',
    responseType: 'blob', // 重要：设置响应类型为 'blob'
    method: 'post',
    data
  })
}
